torch/_dynamo/utils.py

```diff
@@ -219,185 +207,111 @@ def print_time_report() -> None:
+#
+# 1) Logging timings to pt2_compile_events.
+# 2) Logging timings to CompilationMetrics (including dynamo_compile).
+# 3) Chromium events.
+# 4) Storing a record in compilation_time_metrics
+#    For example:
 #
-#   def _foo(...):
-#       with dynamo_timed("_foo"):
-#           ...
+#     def _foo(...):
+#         with dynamo_timed("_foo"):
+#             ...
 #
-# Would show up as an entry in our timing dict:
-# OrderedDict([('_foo', [0.083690, 0.23949, 3.1425e-05])])
-# This is extremely useful for granular debugging.
+#     Would show up as an entry in our timing dict:
+#     OrderedDict([('_foo', [0.083690, 0.23949, 3.1425e-05])])
+#     This is extremely useful for granular debugging.
 #
 # Although it is tempting to use dynamo_timed as a decorator, please do not.
 # In its decorator form it makes cProfile traces less useful as dynamo_timed
 # suddenly becomes a bottleneck for lots of function calls (as only one parent
 # pointer is recorded).
-#
-# For a higher-level mode, pass a phase_name into dynamo_timed
-# phase_names record an extra record into a separate compilation timing structure,
-# one keyed on frame+name rather than function.
-# The frame is incremented outside of this function, in def increment_frame() above.
-# `fwd_only` is used to identify if this phase or function is only called
-# during compiling fwd graphs, e.g, `entire_frame_compile` and `backend_compile`.
-# The other phases (`inductor_compile` and `code_gen`) are called for both fwd and bwd graphs.
-
-
 @contextmanager
 def dynamo_timed(
-    key: str,
-    phase_name: Optional[str] = None,
-    log_pt2_compile_event: bool = False,  # Whether or not to log it to internal pt2 compile event
-    fwd_only: bool = True,
+    event_name: str,
+    # TODO: do we really want event and fn name?
+    fn_name: Optional[str] = None,
+    log_pt2_compile_event: bool = True,
+    metadata: Optional[Dict[str, object]] = None,
+    # TODO: should this be called compilation_metric instead?
+    dynamo_compile_column: Optional[str] = None,
 ):
-    chromium_log: ChromiumEventLogger = get_chromium_event_logger()
-    if key not in compilation_time_metrics:
-        compilation_time_metrics[key] = []
+    """
+    event_name: Event name, e.g., as seen in pt2_compile_events.
+    fn_name: Optional function name for profile record; otherwise use event name.
+    log_pt2_compile_event: Whether to log to internal pt2 compile event.
+    metadata: Extra metadata put in pt2_compile_events.
+    dynamo_compile_column: Name for the dyname_compile column; must be in _us.
+    """
+    # We're standardizing on microseconds for dynamo_compile timings.
+    if dynamo_compile_column is not None:
+        assert dynamo_compile_column.endswith("_us")
 
-    fail_type: Optional[str] = None
-    fail_reason: Optional[str] = None
-    time_spent = float("-inf")
+    mkey = fn_name if fn_name else event_name
+    if mkey not in compilation_time_metrics:
+        compilation_time_metrics[mkey] = []
```

Comment:
nit: `compilation_time_metrics.setdefault(mkey, [])`

Reply:
this thing is existing